docs/ics/ics-xxx-signed-messages.md

```diff
@@ -0,0 +1,150 @@
+# ICS XXX: Cosmos Signed Messages
+
+>TODO: Replace with valid ICS number and possibly move to new location.
+
+## Changelog
+
+## Abstract
+
+Having the ability to sign messages off-chain has proven to be a fundamental aspect
+of nearly any blockchain. The notion of signing messages off-chain has many 
+added benefits such as saving on computational costs and reducing transaction
+throughput and overhead. Within the context of the Cosmos, some of the major
+applications of signing such data includes, but is not limited to, providing a
+cryptographic secure and verifiable means of proving validator identity and
+possibly associating it with some other framework or organization. In addition,
+having the ability to sign Cosmos messages with a Ledger or similar HSM device.
+
+A standardized protocol for hashing, signing, and verifying messages that can be
+implemented by the Cosmos SDK and other third-party organizations is needed. Such a
+standardized protocol subscribes to the following:
+
+* Contains a specification of machine-verifiable and human-readable typed structured data
+* Contains a framework for deterministic and injective encoding of structured data
+* Utilizes cryptographic secure hashing and signing algorithms
+* A framework for supporting extensions and domain separation
+* Is invulnerable to chosen ciphertext attacks
+* Has protection against potentially signing transactions a user did not intend to
+
+This specification is only concerned with the rationale and the standardized
+implementation of Cosmos signed messages. It does **not** concern itself with the
+concept of replay attacks as that will be left up to the higher-level application
+implementation. If you view signed messages in the means of authorizing some
+action or data, then such an application would have to either treat this as 
+idempotent or have mechanisms in place to reject known signed messages.
+
+## Specification
+
+> The proposed implementation is motivated and borrows heavily from EIP-712<sup>1</sup>
+and in general Ethereum's `eth_sign` and `eth_signTypedData` methods<sup>2</sup>.
+
+### Preliminary
+
+We will a have Cosmos SDK message signing protocol that consists of using `SHA-256`,
+as the hashing algorithm and `secp256k1` as the signing algorithm.
```

Comment:
I think we can more or less specify the format independently of the signing algorithm (just parameterize it as some `S`, with `sign` and `verify`), then instantiate the format for whatever signing algorithms we need (initially secp256k1, later ed25519).

Reply:
Not sure I 100% understand what you mean, but I updated the preliminary to be inline with what I think you're suggesting.